src/lib/progress.c

```diff
@@ -153,20 +153,20 @@ static void progress_spinner_print_end(void)
 
 void progress_set_spinner(bool status_flag)
 {
-	if (log_get_level() == LOG_DEBUG || !isatty(fileno(stdout))) {
+	if (log_get_level() == LOG_DEBUG) {
 		return;
 	}
 
 	if (status_flag) {
 		if (progress_function) {
 			progress_report(-1, 100);
-		} else {
+		} else if (isatty(fileno(stdout))) {
```

Comment:
can you do the same on the case of --debug?

if you run

    swupd diagnose --debug --json-output

we should print the -1 too

Reply:
Done